oracle-uri-proposal.md

```diff
@@ -0,0 +1,179 @@
+Path segments that represent proper nouns of acronyms should be capitalised as they would be written, otherwise they should be lower `kebab-case`.
+
+Each following segment should narrow the event id until it focuses on a particular event in space and time. 
+Oracles are free to define the structure for this themselves with the following considerations:
+
+1. The event type of the event may require the path has a certain structure.
+2. Rules and conventions may be attached to namespaces and in general oracles should try and follow them.
+
+### Announcement
+
+Oracles must announce their intention to attest to the outcome of an event ahead of time.
+This is done by signing a message which includes the event id.
+Additionally it usually involves committing to some auxiliary cryptographic data (e.g. nonces) which allows the clients to _anticipate_ the final signature for any message.
+The announcement signature is made by suffixing the event id with `#` and appending the cryptographic data.
+In addition, other metadata may be added here such as the time the oracle expects to be able to attest to the event.
+
+_how to encode the data is left for a future revision_
+
+### Attestation
+
+When attesting to a particular outcome for the event the oracle appends `=` and the outcome text onto the event id and signs the result.
+For example, if the outcome of `/NBA/s2019/2020-08-11/PHI_PHX?vs` is `PHX_win` then the oracle signs `/NBA/s2019/2020-08-11/PHI_PHX?vs=PHX_win`.
+
+### Event Types
+
+The _event type_ occupies the _query string_ part of the URI.
+Event types will be introduced in their own separate RFCs and all event types mentioned here are just an example.
+The event type is a kebab-case identifier which may be followed by _parameters_ enclosed in parenthesis.
+The parameters are a list of comma separated values.
+Each event type defines its parameters and together they determine the outcomes for the event.
+
+The event type will usually define how to interpret the preceding path segments.
+For example, for a `vs` event type may require that the final path segment be in the form `<team1>_<team2>` or a `weather` event type may require the last two segments to be in the form `<location>/<date>`.
+
+We explicitly disallow the oracle to communicate what the outcomes are for a particular event explicitly.
+Outcomes are always determined implicitly from the event type.
+This is done to avoid oracles independently creating events with similar but not precisely the same semantics and structure.
+For example, one oracle might have a weather event that has outcomes `sunny`, `cloudy`, `rainy` while another may have `light-rain` in addition.
+The hope is that this restriction will encourage the public collaboration on new event types and rich client support for them.
```

Comment:
I think that there should be a built-in (so as to encourage use) versioning of event types in the URI.

Reply:
I would hope explicit versions will not be necessary. If we screw up specifying `weather`, you can always make `weather2` or perhaps `weather.2` if we want to make visually clear it's a version.

(as side not weather is a bad example since I think it is not the type of thing people will actually use this for)